### PR TITLE
UNR-1738 Initial AlwaysRelevant support

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Added AlwaysRelevant component that gets attached to Actors that have bAlwaysRelevant set. This is currently only parsed at entity creation and won't get dynamically toggle if the variable is. 

#### Primary reviewers
@Vatyx @yumnuska

### CODE REVIEW

File: SpatialGDK/Extras/schema/relevant.schema [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
```
Comment (Vatyx) [inline SpatialGDK/Extras/schema/relevant.schema]: the schema should probably be named AlwaysRelevant.schema?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 2]
```diff
@@ -136,6 +137,11 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 		ComponentWriteAcl.Add(SpatialConstants::HEARTBEAT_COMPONENT_ID, OwningClientOnlyRequirementSet);
 	}
 
+	if (Actor->bAlwaysRelevant)
+	{
+		ComponentWriteAcl.Add(SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Any reason we couldn't add this to the WriteAcl all the time?

File: SpatialGDK/Extras/schema/relevant.schema (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Extras/schema/relevant.schema]: I was thinking we could put other Unreal relevant components in here - OwnerOnly etc.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Good point

File: SpatialGDK/Source/SpatialGDK/Public/Schema/AlwaysRelevant.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace SpatialGDK
+{
+
+struct AlwaysRelevant : Component
+{
+	static const Worker_ComponentId ComponentId = SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID;
+
+	AlwaysRelevant() = default;
+	AlwaysRelevant(const Worker_ComponentData& Data)
```
Comment (yumnuska) [inline SpatialGDK/Source/SpatialGDK/Public/Schema/AlwaysRelevant.h]: Probably want an explicit ctor

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 4]
```diff
@@ -355,22 +355,24 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint() const
...
+QueryConstraint InterestFactory::CreateAlwaysRelevantConstraint() const
 {
-	QueryConstraint SingletonConstraint;
+	QueryConstraint AlwaysRelevantConstraint;
 
-	Worker_ComponentId SingletonComponentIds[] = {
+	Worker_ComponentId ComponentIds[] = {
 		SpatialConstants::SINGLETON_COMPONENT_ID,
-		SpatialConstants::SINGLETON_MANAGER_COMPONENT_ID };
+		SpatialConstants::SINGLETON_MANAGER_COMPONENT_ID,
```
Comment (yumnuska) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: maybe not worth it now, but we discussed removing the manager from this list a while back.

File: SpatialGDK/Extras/schema/relevant.schema (hunk 1 above)
Comment (yumnuska) [reply SpatialGDK/Extras/schema/relevant.schema]: I like that plan.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp (hunk 4 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: Yep, we can do that when we finally remove the streaming query also.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 5]
```diff
@@ -136,6 +137,8 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 		ComponentWriteAcl.Add(SpatialConstants::HEARTBEAT_COMPONENT_ID, OwningClientOnlyRequirementSet);
 	}
 
+	ComponentWriteAcl.Add(SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Do we want to always add the ACL?

**Pull request merged**